features/teacher.feature

```diff
@@ -28,6 +28,58 @@ Scenario: Logging in as a teacher should be able to edit their info
   Then  the "First Name" field should contain "Joe"
   Then  there is a TEALS email
 
+Scenario: Logged in teacher can fill a new form with their info
+  Given the following schools exist:
+  |       name      |     city     |  state  |            website            |
+  |   UC Berkeley   |   Berkeley   |   CA    |   https://www.berkeley.edu    |
+  Given the following teachers exist:
+  | first_name | last_name | admin | email                    | school      |
+  | Joseph     | Mamoa     | false | testteacher@berkeley.edu | UC Berkeley |
+  Given I have a teacher email
+  Given I am on the BJC home page
+  And   I follow "Log In"
+  Then  I can log in with Google
+  When  I go to the new teachers page
+  And   I enter my "First Name" as "Joe"
+  And   I enter my "Last Name" as "Mamoa"
+  And   I enter my "School Email" as "testteacher@berkeley.edu"
+  And   I enter my "Snap! Username" as "alonzo"
+  And   I set my status as "I am teaching BJC as an AP CS Principles course."
+  And   I set my education level target as "High School"
+  And   I enter my "School Name" as "Cupertino High School"
+  And   I enter my "City" as "Cupertino"
+  And   I select "CA" from "State"
+  And   I enter my "School Website" as "https://chs.fuhsd.org"
+  And   I press "Submit"
+  Then  I see a confirmation "Successfully updated your information"
+  And   I am on the edit page for Joe Mamoa
+
+  Scenario: Logged in teacher cannot change Snap from new form path
+  Given the following schools exist:
+  |       name      |     city     |  state  |            website            |
+  |   UC Berkeley   |   Berkeley   |   CA    |   https://www.berkeley.edu    |
+  Given the following teachers exist:
+  | first_name | last_name | admin | email                    | school      |
+  | Joseph     | Mamoa     | false | testteacher@berkeley.edu | UC Berkeley |
+  Given I have a teacher email
+  Given I am on the BJC home page
+  And   I follow "Log In"
+  Then  I can log in with Google
+  When  I go to the new teachers page
+  And   I enter my "First Name" as "Joe"
+  And   I enter my "Last Name" as "Mamoa"
+  And   I enter my "School Email" as "testteacher@berkeley.edu"
+  And   I enter my "Snap! Username" as "not_alonzo"
+  And   I set my status as "I am teaching BJC as an AP CS Principles course."
+  And   I set my education level target as "High School"
+  And   I enter my "School Name" as "Cupertino High School"
+  And   I enter my "City" as "Cupertino"
+  And   I select "CA" from "State"
+  And   I enter my "School Website" as "https://chs.fuhsd.org"
+  And   I press "Submit"
+  Then  I see a confirmation "Failed to update your information"
+  And   I am on the edit page for Joseph Mamoa
+
```

Comment:
I am not sure tests actually might suffice to capture the intended functionality because in the form, the fields "emails" and "snap username" are disabled by javascript, but I don't know if Cucumber can bypass that or something.

Reply:
So it is actually in the html. I manually disabled the html readonly on the form, and it seems it is working as intended.

Reply:
The other thing too is this is testing the new teacher form (where the fields are not disabled) and not the update teacher form so I was just checking an existing teacher couldn't bypass the block by trying to update their info in the new teacher form

Reply:
If there is a need to test javascript functionality with cucumber, you just need to make sure you have an appropriate driver installed to do so. Selenium + geckodriver for Mozilla has worked for me quite well, and you can run it either with head or headless based on whether you want to see the actions being performed visually, which can be helpful when combined with byebug to see what's working and what's not when it comes to javascript.